articles/153_roslaunch_remote_launch.md

```diff
@@ -0,0 +1,185 @@
+# Multi-Machine and Remote Launching
+## Goals
+- Connect to a remote host and run nodes on it
+    - Ensure this capability does not compromise security
+- Support arbitrary remote execution methods
+- Monitor the status of and manage lifecycles of nodes across systems
+- Shutdown nodes remotely
+- Ability to detach the machine used to launch a system and have the system continue running
+    - Be able to reconnect to the system to monitor and manage nodes running remotely
+
+
+To achieve our goals, this update will consist of two primary components: an ExecuteRemote action, and a LaunchServiceNode.
+
+The ExecuteRemote action will build off of the refactor he is working on described here: https://github.com/ros2/design/pull/272
+Since we don't want to limit remote execution to a single protocol, ExecuteRemote will be a base class and we will create an implementation using SSH which we will call ExecuteRemoteSSH.
+Remote machines will be described by a Machine class which will also be a base class for protocol-specific implementations.
+Together these components will provide the means to execute processes remotely and should cover most basic use cases.
+
+The LaunchServiceNode(s) will expose ROS2 services and topics for handling startup and shutdown of processes on remote machines.
+This component won't be necessary for basic execution of remote nodes, but would allow remote system to persist beyond the life of the initial LaunchService.
```

Comment:
I am not clear on this use case, could you elaborate a little bit? especially on the difference from `LifecycleNode` except process statistics?

Reply:
The design is intended to allow you to run a node pretty much like they currently would run, that is, bound to the lifespan of the launch process itself. When the launch process dies, the SSH connection dies, and anything still running on that SSH terminal dies with it.

The `LaunchServiceNode` is intended to work a little bit differently. It would be launched from the SSH terminal, but then detach and run without requiring the SSH connection to remain up. However, by doing that you lose the ability to directly kill a process if you actually want it to stop. Therefore, the `LaunchServiceNode` would provide a way to go back and ask for particular nodes on that server to be started/stopped ad hoc.

This is certainly somewhat similar to some of the functionality envisioned by `LifecycleNode`s, but would allow a way for us to remotely launch and manage nodes which were not natively designed to support the lifecycle feature. Lifecycle would provide nearly all the functionality we would need for this type of behavior, but should the node crash, Lifecycle won't let us bring it back for a full restart, and we also would be lacking other out-of-band control to deal with things such as unresponsive nodes.

(Obviously, these same arguments also apply to the `LaunchServiceNode` itself, so that particular entity should be as clean as possible to reduce the possibility of it getting into an unmanageable state itself.)

Reply:
Question to explicitly state in the doc: Who starts the LaunchServiceNode? How is its lifecycle expected to be managed?